docs/tutorial/part04.rst

```diff
@@ -2,5 +2,495 @@
+
+    countries = db.cypher_query(
+        '''
+        MATCH (n)
+        WHERE NOT n.countries CONTAINS ';'
+        RETURN DISTINCT n.countries AS countries
+        '''
+        )[0]
+
+    jurisdictions = db.cypher_query(
+        '''
+        MATCH (n)
+        RETURN DISTINCT n.jurisdiction AS jurisdiction
+        '''
+    )[0]
+
+    data_sources = db.cypher_query(
+        '''
+        MATCH (n)
+        RETURN DISTINCT n.sourceID AS dataSource
+        '''
+    )[0]
+
+
+With the results we will make sorted lists of COUNTRIES, JURISDICTIONS and DATASOURCE::
+
+    COUNTRIES = sorted([country[0] for country in countries])
+    JURISDICTIONS = sorted([jurisdiction[0] for jurisdiction in jurisdictions if isinstance(jurisdiction[0], str)])
+    DATASOURCE = sorted([data_source[0] for data_source in data_sources if isinstance(data_source[0], str)])
+
+In the ``utils.py``, import COUNTRIES, JURISDICTIONS, DATASOURCE::
+
+    from paradise_papers_search.constants import COUNTRIES, JURISDICTIONS, DATASOURCE
+
+Then create ``fetch_countries``, ``fetch_jurisdictions`` and ``fetch_data_source`` helpers::
+
+    def fetch_countries():
+        return COUNTRIES
+
+
+    def fetch_jurisdictions():
+        return JURISDICTIONS
+
+
+    def fetch_data_source():
+        return DATASOURCE    
+
+We created the ``constants.py`` module because we want to make the cypher queries once. Not each
+time we call ``fetch_countries``, ``fetch_jurisdictions`` or ``fetch_data_source`` helpers.
+
+Since these queries might take some time to execute, we want them ready at the start of the
+application. We can do that by executing the ``constants.py`` code.
+
+Open the ``fetch_api/app.py`` module. Add a new method to the Django application class with the name ``ready`` and import the ``constants.py`` module. That will be enough to initialize 
+COUNTRIES, JURISDICTIONS and DATASOURCE constants.
+
+Here how the ``fetch_api/app.py`` would look like::
+
+    from django.apps import AppConfig
+
+    class FetchApiConfig(AppConfig):
+        name = 'fetch_api'
+
+        def ready(self):
+            from paradise_papers_search import constants
+
+Using the search utils
+======================
+
+To use the search utils, we just need to import them into the module they will be used. In this
+case, we will need to import them into the ``fetch_api/views.py`` module. Later they will be used
+to create our application views.
+
+Here the import statement, place this code in the ``fetch_api/views.py`` module::
+
+    from .utils import (
+        count_nodes,
+        fetch_nodes,
+        fetch_node_details,
+        fetch_countries,
+        fetch_jurisdictions,
+        fetch_data_source,
+    )
+
+In the next section, you will build the rest of this code.
+
+Testing utils in the console
+----------------------------
+
+Just like we did models, we can the utils in the project python interpreter and play around.

```

Comment:
Amend:

Just like we did **with** models, we can **import** the utils in the project python interpreter and play around.